nb-apis/adapters-api/src/main/java/io/nosqlbench/adapters/api/activityimpl/uniform/ConcurrentIndexCache.java

```diff
@@ -0,0 +1,137 @@
+package io.nosqlbench.adapters.api.activityimpl.uniform;
+
+/*
+ * Copyright (c) 2022 nosqlbench
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
```

Comment:
Add safety check for allocating > heap

Reply:
I added a warning for each time the cache is resized to over 1M elements, which is 8MB, so if a user is seeing this, they will have multiple warnings before it causes an issue.

Reply:
I changed this to have an explicit max index, which throws an error if exceeded, but which is configurable by the caller. If defaults to 1000000 if not provided.